lib/task_helpers/metrics.rb

```diff
@@ -0,0 +1,49 @@
+module TaskHelpers
+  class Metrics
+    class << self
+      def per_tenant
+        column_names = %w[
+          tenant
+          portfolio_count
+          product_count
+          portfolio_share_count
+        ].freeze
+        data = [column_names.collect(&:titleize)]
+
+        Tenant.order(:external_tenant).each do |current_tenant|
+          ActsAsTenant.with_tenant(current_tenant) do
+            next if tenant_empty?
+
+            data << column_names.collect { |column_name| send(column_name) }
+          end
+        end
+
+        data
+      end
+
+      def tenant
+        ActsAsTenant.current_tenant.external_tenant
+      end
+
+      def portfolio_count
+        Portfolio.count
+      end
+
+      def portfolio_share_count
+        Portfolio.count do |portfolio|
+          portfolio.metadata['statistics']['shared_groups'] > 0
+        end
+      end
+
+      def product_count
+        PortfolioItem.count
+      end
+
+      private
+
+      def tenant_empty?
```

Comment:
This could probably have a better name, since the tenant isn't really the thing that is empty or missing, it just doesn't have any orders or portfolios. Since you're using it in the context of `next if <blank>`, maybe it should be something like `next unless tenant_has_data?` (then of course the logic would have to change, but it reads more like exactly what's going on).